### PR TITLE
handle flat rolling (no dim specified) T36264909

### PR DESCRIPTION
**Pull request by nairbv**
update roll to behave as in numpy.roll when dimension to roll not specified.

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -4692,7 +4692,9 @@ def parse_kwargs(desc):
 roll(input, shifts, dims) -> Tensor
```
Comment (gchanan) [inline torch/_torch_docs.py]: the string here should be updated to dims=None.

File: test/test_torch.py [hunk 2]
```diff
@@ -7313,6 +7313,10 @@ def test_roll(self):
             self.assertEqual(expected, rolled,
                              "non contiguous tensor rolled to {} instead of {} ".format(rolled, expected))
 
+            # test roll with no dimension specified
+            expected = numbers.roll(1, 0).view(2, 4)
```
Comment (gchanan) [inline test/test_torch.py]: this test looks good -- can you also add an empty test as you did before?

File: test/test_torch.py [hunk 3]
```diff
@@ -7313,6 +7313,10 @@ def test_roll(self):
...
+            self.assertEqual(expected, data.roll(1), "roll with no dims should flatten and roll.")
+
```
Comment (gchanan) [inline test/test_torch.py]: also test explicitly passing dims=None.

File: aten/src/ATen/native/TensorTransformations.cpp [hunk 4]
```diff
@@ -58,19 +58,24 @@ Tensor flip_cpu(const Tensor& self, IntList dims) {
...
+  if (self.numel() == 0) {
     return self.clone();
   }
   int64_t dim = dims[0];
   int64_t size = self.size(dim);
   int64_t start = (size - shifts[0]) % size;
   // Behavior of % is different in C++ vs Python for negative numbers. This
   // corrects the difference.
-  if( start < 0 ) start = start + size;
+  if (start < 0)
```
Comment (gchanan) [inline aten/src/ATen/native/TensorTransformations.cpp]: if you do this can you surround it with {} -- too easy to add something and forget.

**Review by gchanan — APPROVED** [review]
looks good, just nits.

File: test/test_torch.py (hunk 2 above)
Comment (nairbv) [reply test/test_torch.py]: added

File: test/test_torch.py (hunk 3 above)
Comment (nairbv) [reply test/test_torch.py]: done

**Pull request closed without merging**